src/Cache.php

```diff
@@ -55,38 +73,39 @@ public function __construct(\Psr\SimpleCache\CacheInterface $handler = null)
      * @param mixed $key the key to be normalized
      * @return string the generated cache key
      */
-    private function buildKey($key): string
+    private function normalizeKey($key): string
     {
-        if (\is_string($key)) {
-            return ctype_alnum($key) && mb_strlen($key, '8bit') <= 32 ? $key : md5($key);
+        if (!$this->keyNormalization) {
+            $normalizedKey = $key;
+        } elseif (\is_string($key)) {
+            $normalizedKey = ctype_alnum($key) && mb_strlen($key, '8bit') <= 32 ? $key : md5($key);
+        } else {
+            $normalizedKey = $this->keyPrefix . md5(json_encode($key));
         }
-        return md5(json_encode($key));
+
+        return $this->keyPrefix . $normalizedKey;
```

Comment:
Normalization and prefixing are two separate actions. `normalizeKey()` name suggests that normalization is done inside so either it's not a good name (`buildKey()` is fine in this case) or there should be two methods.